src/js/launcher.js

```diff
@@ -0,0 +1,115 @@
+"use strict";
+var fluid  = require("infusion");
+var gpii   = fluid.registerNamespace("gpii");
+var kettle = fluid.registerNamespace("kettle");
+
+fluid.require("%kettle/lib/KettleConfigLoader.js");
+
+var yargs   = require("yargs");
+var path    = require("path");
+var process = require("process");
+
+// A global resolver that can be used to pull raw environment variables from configuration blocks.
+fluid.defaults("gpii.launcher.resolver", {
```

Comment:
To complete our "pull" support, we should have parallels of this grade for command line options and JSON/JSON5 files held in the filesystem.